lib/wallet/wallet.js

```diff
@@ -1644,66 +1644,123 @@ class Wallet extends EventEmitter {
 
-    if (!rules.hasRollout(nameHash, height, network))
-      throw new Error(`Name not yet available: "${name}".`);
+      // TODO: Handle expired behavior.
+      if (rules.isReserved(nameHash, height, network)) {
+        errors.push({name: name, error: `Name is reserved: "${name}".`});
+        continue;
+      }
 
-    let ns = await this.getNameState(nameHash);
+      if (!rules.hasRollout(nameHash, height, network)) {
+        errors.push({name: name, error: `Name not yet available: "${name}".`});
+        continue;
+      }
 
-    if (!ns)
-      ns = await this.wdb.getNameStatus(nameHash);
+      let ns = await this.getNameState(nameHash);
+      if (!ns) {
+        ns = await this.wdb.getNameStatus(nameHash);
+      }
 
-    ns.maybeExpire(height, network);
+      ns.maybeExpire(height, network);
 
-    const state = ns.state(height, network);
-    const start = ns.height;
+      const state = ns.state(height, network);
+      const start = ns.height;
 
-    if (state !== states.OPENING)
-      throw new Error(`Name is not available: "${name}".`);
+      if (state !== states.OPENING) {
+        errors.push({name: name, error: `Name is not available: "${name}".`});
+        continue;
+      }
 
-    if (start !== 0 && start !== height)
-      throw new Error(`Name is already opening: "${name}".`);
+      if (start !== 0 && start !== height) {
+        errors.push({name: name, error: `Name is already opening: "${name}".`});
+        continue;
+      }
 
-    const addr = await this.receiveAddress(acct);
+      const addr = await this.receiveAddress(acct);
 
-    const output = new Output();
-    output.address = addr;
-    output.value = 0;
-    output.covenant.type = types.OPEN;
-    output.covenant.pushHash(nameHash);
-    output.covenant.pushU32(0);
-    output.covenant.push(rawName);
+      const output = new Output();
+      output.address = addr;
+      output.value = 0;
+      output.covenant.type = types.OPEN;
+      output.covenant.pushHash(nameHash);
+      output.covenant.pushU32(0);
+      output.covenant.push(rawName);
 
-    const mtx = new MTX();
-    mtx.outputs.push(output);
+      if (await this.txdb.isCovenantDoubleOpen(output.covenant)) {
+        errors.push({name: name, error: `Already sent an open for: ${name}.`});
+        continue;
+      }
 
-    if (await this.txdb.isDoubleOpen(mtx))
-      throw new Error(`Already sent an open for: ${name}.`);
+      mtx.outputs.push(output);
+    }
 
-    return mtx;
+    const isAllError = (names.length === errors.length);
+    return { mtx: mtx, errors: errors, isAllError: isAllError};
+  }
+
+  /**
+   * Create and finalize a batch open
+   * MTX without a lock.
+   * @param {Array} names
+   * @param {Boolean} force
+   * @param {Object} options
+   * @returns {MTX}
+   */
+
+  async _createBatchOpen(names, force, options) {
+    const acct = options ? options.account || 0 : 0;
+    const {mtx, errors, isAllError} = await this
+    .makeBatchOpen(names, force, acct);
```

Comment:
The formatting in this function is strange, can you resolve all these indentation issues? 1741 has 3 spaces, this line 1739 has no indentation, etc. I am not sure why the linter is not catching this

Reply:
respect for pointing out the line numbers ...

Reply:
fixed